File: quarkc/test/lib/callable_ffi_test.js
exports.check = function () {
    var callable = require(".").callable_test;
    if (6 !== callable.smug(function (x) { return 2*x; }, 3)) {
        return false;
    }
    // undefined is converted into null:
    return null === callable.smug(function (x) { return undefined; }, null);
};

File: quarkc/test/lib/callable_ffi_test.rb
require("callable_test")

module CallableFFI

  class Foo
    def foo(x)
      2*x
    end
  end

  def check()
    return 6 == Quark::CallableTest::smug(lambda {|x| 2*x}, 3) &&
           6 == Quark::CallableTest::smug(Proc.new {|x| 2*x}, 3) &&
           6 == Quark::CallableTest::smug(Foo.new.method(:foo), 3)
  end

  module_function :check

end
